dask/order.py

```diff
@@ -349,6 +351,14 @@ def dependencies_key(x):
         while item in result:
             item = init_stack_pop()
 
+    # Check task annotations
+    for k, v in dsk.items():
+        if istask(v):
+            f = v[0]  # We prioritize based on the first task function
```

Comment:
What if the task is fused?  It seems like this approach, while simple, may silently ignore user annotations if they have gone through task fusion (which is common).

Reply:
True, in this case we only check the first function. We could use [`functions_of()`](https://github.com/dask/dask/blob/1f87ae6329c3ae1c6f3f31945f891c33642c64ba/dask/optimization.py#L352) to check all nested function with the extra overhead that entails. 
Alternately as @sjperkins suggest, the fusers could also check the priority annotation and act accordingly.